### PR TITLE
Shiny create bugs

### LINKED ISSUE #956: Shiny create needs dir flag
For writing tests for `shiny create` we intend to use a temp directory and it would be easy if `shiny create` allows user to pass a dir or `-d` flag that user can specify the directory of the app.

The tests will invoke `shiny create` via the command line/shell to test true E2E and then start the app to check for console errors.

@gshotwell

### LINKED ISSUE #957: Shiny create: Ask users to install requirements.txt
When using `shiny create` for apps like `Dashboard` which uses extra python packages, even though we copy over the `requirements.txt` during the app creation, we need to let the user name to install them since a lot of new users might not have them installed. 

Found by @dfalty during the bug bash

### LINKED ISSUE #958: Shiny create: Basic App creation overrides files.
When creating a `Basic App` in the directory with another shiny app, the files get overridden. When creating a `Dashboard` or something else, I get the proper error message. `Error: Can't create new files because the following files already exist in the destination directory:`

Steps:
1. `shiny create`
2. Select `Basic App`
3. Select `Yes` for Shiny Express
4. pick a destination directory
5. Edit the app.py file that is created
6. `shiny create` again
7. Select `Basic App`
8. Select `Yes` for Shiny Express
9. pick the same destination directory as # 4
10. See the edits for app.py were overridden

### LINKED ISSUE #960: README: Shiny create section shows older way of shiny templates
If you refer to the README in the homepage for the repo, it shows
py-shiny_README_md_at_main_·_posit-dev_py-shiny
which is not supported with the latest version of shiny

### LINKED ISSUE #961: using the `-t` and `-g` flags prefers the the `-g` flag
in the help text for `shiny create` can we mention that when using the `-t` and `-g` flags together, the `-g` flag takes precedence?

### LINKED ISSUE #962: Shiny Create: React Component fails on `npm run build`
Shiny version: `0.6.1.1`
Python version: `3.11.6`
Node version: `v21.2.0`
npm version: `10.2.3`

To reproduce:
1. Run `shiny create`
2. Select `Custom Javascript Component -> React component`.
3. Name the directory `react-component`
4. `npm install` to install all node packages
5. `npm run build`

See the error below.

```
react-component $ npm run build  

> react-component-pkg-template@1.0.0 build
> esbuild srcts/index.tsx --bundle --outfile=react_component/distjs/index.js

✘ [ERROR] Could not resolve "@shiny-helpers/react"

    srcts/index.tsx:4:48:
      4 │ import { makeReactInput, makeReactOutput } from "@shiny-helpers/react";
        ╵                                                 ~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@shiny-helpers/react" as external to exclude it from the bundle, which will
  remove this error and leave the unresolved path in the bundle.

1 error
```

### PR DESCRIPTION
**Pull request by gshotwell**
This closes a bunch of issues from the QA bug bash. 

Fixes #961
Fixes #960 
Fixes #958 
Fixes #957 
Fixes #956 
Fixes #962

### CODE REVIEW

File: shiny/_custom_component_template_questions.py [hunk 1]
```diff
@@ -1,10 +1,27 @@
 import re
+from importlib import util
```
Comment (schloerke) [inline shiny/_custom_component_template_questions.py:2]: Standard package. No need for new dependency

File: tests/pytest/test_shiny_create.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    else:
+        assert os.path.isfile(f"{dest_dir}/pyproject.toml")
+
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(cmd, check=True)
+
+    shutil.rmtree(dest_dir)
+
+
+@pytest.mark.parametrize("app_template", ["basic-app", "dashboard", "multi-page"])
```
Comment (schloerke) [inline tests/pytest/test_shiny_create.py:39]:
```suggestion
# TODO-karan; Integrate all tests below with _examples_ testing and check for JS errors / output errors. 
@pytest.mark.parametrize("app_template", ["basic-app", "dashboard", "multi-page"])
```

File: shiny/_main.py [hunk 3]
```diff
@@ -513,18 +513,38 @@ def try_import_module(module: str) -> Optional[types.ModuleType]:
     "-g",
     help="The GitHub URL of the template sub-directory. For example https://github.com/posit-dev/py-shiny-templates/tree/main/dashboard",
 )
+@click.option(
+    "--directory",
```
Comment (schloerke) [inline shiny/_main.py:517]: Up to you... but py-shinylive has `--dir` for output directory values

We'd only have to update the `create()` method

**Review by schloerke — APPROVED** [review]
Thank you for adding CLI tests!

@karangattu can add some extra execution tests on top of them in a followup PR. 😃

File: shiny/_main.py (hunk 3 above)
Comment (gshotwell) [reply shiny/_main.py:517]: Oh yeah may as well keep them consistent

**Pull request merged**